inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,16 +26,16 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
     file_name: str
     sources: list[str] = field(default_factory=lambda: [])
+    gpg_key_uri: Optional[str] = field(default=None)
+    gpg_key_name: Optional[str] = field(default=None)
```

Comment:
**Suggestion:** Consider using a more specific type than list for sources to ensure immutability, such as tuple. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
@dataclass(kw_only=True, frozen=True)
class ApplicationAddSourceParameters:
    file_name: str
    sources: tuple[str, ...] = field(default_factory=tuple)
    gpg_key_uri: Optional[str] = field(default=None)
    gpg_key_name: Optional[str] = field(default=None)
```